### PR TITLE
 PDD module upgrade v2

### PR DESCRIPTION
**Pull request by leeinsunny**
- Change the pattern of the resident registration number to reflect the date of birth
	- Use the DateTime module to handle the date in the resident registration number
	- Resident registration numbers on valid dates are filtered as '*', but resident registration numbers on invalid dates are not filtered.
- Implement address patterns as regular expressions so that they can be detected by entering searchable addresses
	- We do not filter until ~si, ~do
	- Filter with one "*" mark after ~si and ~do
-
image

image

### CODE REVIEW

File: dt-demo-be/modules/pdd/run.py [hunk 1]
```diff
@@ -57,9 +72,23 @@ def run(fileName):
...
-                    if re.search(pattern_dict[pattern]["pattern"], target):
+                   match = re.search(pattern_dict[pattern]["pattern"], target)  
+                   if match:
                         print(f'{pattern_dict[pattern]["type"]} detected on target sentence:')
                         print(f"{target}")
+                        # 주민등록번호 패턴이라면
+                        if "res_num" in pattern:
+                            year, month, day = int(match.group(1)), int(match.group(2)), int(match.group(3))
+                            # 연도 확인
+                            if match.group(4) in ['1', '2']:  # 주민번호 뒷자리가 1이나 2이면 1900년대 생
```
Comment (doorooful) [inline dt-demo-be/modules/pdd/run.py:83]: Can you use ternary operator here?
i.e)
yr += 1900 if gender<3 else 2000
(please edit gender part)

**Review by doorooful — CHANGES_REQUESTED** [review]
Can you update result screenshot as to show only updated functions?
Thanks for your effort!

**Review by yoouung — APPROVED** [review]
looks good to me :)

File: dt-demo-be/modules/pdd/run.py (hunk 1 above)
Comment (leeinsunny) [reply dt-demo-be/modules/pdd/run.py:83]: image
image
image
image

**Review by doorooful — APPROVED** [review]
Thanks☺️

**Pull request merged**